programs/ido-program/programs/ido-program/src/lib.rs

```diff
@@ -0,0 +1,438 @@
+    #[account(seeds = [pool.native_mint.as_ref()], bump = pool.bump)]
+    pool_signer: AccountInfo<'info>,
+
+    #[account(
+        mut,
+        constraint = redeemable_mint.mint_authority == COption::Some(*pool_signer.key)
+    )]
+    pub redeemable_mint: Account<'info, Mint>,
+
+    #[account(mut, constraint = pool_fiat.mint == *fiat_mint.to_account_info().key)]
+    pub fiat_mint: Account<'info, Mint>,
+
+    #[account(mut, constraint = pool_fiat.owner == *pool_signer.key)]
+    pub pool_fiat: Account<'info, TokenAccount>,
+
+    #[account(mut)]
+    pub authority: Signer<'info>,
+
+    #[account(mut, constraint = investor_fiat.owner == *authority.key)]
+    pub investor_fiat: Account<'info, TokenAccount>,
+
+    #[account(mut, constraint = investor_redeemable.owner == *authority.key)]
+    pub investor_redeemable: Account<'info, TokenAccount>,
+
+    #[account(constraint = token_program.key == &token::ID)]
+    pub token_program: Program<'info, Token>,
+
+    pub clock: Sysvar<'info, Clock>,
+}
+
+#[derive(Accounts)]
+pub struct ExchangeRedeemableForNative<'info> {
+    #[account(has_one = redeemable_mint, has_one = pool_native)]
+    pub pool: Box<Account<'info, PoolAccount>>,
+
+    /// CHECK: This is not dangerous
+    #[account(seeds = [pool.native_mint.as_ref()], bump = pool.bump)]
+    pool_signer: AccountInfo<'info>,
+
+    #[account(
+        mut,
+        constraint = redeemable_mint.mint_authority == COption::Some(*pool_signer.key)
+    )]
+    pub redeemable_mint: Account<'info, Mint>,
+
+    #[account(mut, constraint = pool_native.owner == *pool_signer.key)]
+    pub pool_native: Account<'info, TokenAccount>,
+
+    #[account(mut)]
+    pub authority: Signer<'info>,
+
+    #[account(mut, constraint = investor_native.owner == *authority.key)]
+    pub investor_native: Account<'info, TokenAccount>,
+
+    #[account(mut, constraint = investor_redeemable.owner == *authority.key)]
+    pub investor_redeemable: Account<'info, TokenAccount>,
+
+    #[account(constraint = token_program.key == &token::ID)]
+    pub token_program: Program<'info, Token>,
+
+    pub clock: Sysvar<'info, Clock>,
+}
+
+#[derive(Accounts)]
+pub struct WithdrawPoolFiat<'info> {
+    #[account(has_one = pool_fiat)]
+    pub pool: Box<Account<'info, PoolAccount>>,
+
+    ///CHECK: This is not dangerous
+    #[account(seeds = [pool.native_mint.as_ref()], bump = pool.bump)]
+    pub pool_signer: AccountInfo<'info>,
+
+    #[account(mut, constraint = pool_fiat.mint == *fiat_mint.to_account_info().key)]
+    pub fiat_mint: Account<'info, Mint>,
+
+    #[account(mut, constraint = pool_fiat.owner == *pool_signer.key)]
+    pub pool_fiat: Account<'info, TokenAccount>,
+
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    #[account(mut)]
+    pub creator_fiat: Account<'info, TokenAccount>,
+
+    #[account(constraint = token_program.key == &token::ID)]
+    pub token_program: Program<'info, Token>,
+
+    pub clock: Sysvar<'info, Clock>,
```

Comment:
Get the `Clock` from runtime with `Clock::get()?` instead of passing as an account
```suggestion
```